### PR TITLE
Updates CRA from v4 to v5

### PR DESCRIPTION
**Pull request by shayneczyzewski**
# Description

Updates Create React App from v4 to v5. It is pretty straightforward:
- https://create-react-app.dev/docs/updating-to-new-releases/
- https://github.com/facebook/create-react-app/blob/main/CHANGELOG.md#migrating-from-40x-to-500

The only drag is we do have a couple of deprecation warnings (however, this is a known CRA5 issue that will hopefully be fixed soon: https://github.com/facebook/create-react-app/issues/11860):

```bash
Web app (stderr): (node:10363) [DEP_WEBPACK_DEV_SERVER_ON_AFTER_SETUP_MIDDLEWARE] DeprecationWarning: 'onAfterSetupMiddleware' option is deprecated. Please use the 'setupMiddlewares' option.
Web app (stderr): (Use `node --trace-deprecation ...` to show where the warning was created)
Web app (stderr): (node:10363) [DEP_WEBPACK_DEV_SERVER_ON_BEFORE_SETUP_MIDDLEWARE] DeprecationWarning: 'onBeforeSetupMiddleware' option is deprecated. Please use the 'setupMiddlewares' option.
Web app (stderr): Starting the development server...
```

If we really wanted to hide it, we could filter the web app output.

Fixes most deprecation warnings from: #181 (The only ones left are stable, w3c-hr-time, and svgo. But down from 21 before upgrade. And far fewer npm audit errors.)

The goal for this is to make it easier to add Tailwind support as a next step (https://tailwindcss.com/docs/guides/create-react-app).

## Type of change

Please select the option(s) that is more relevant.

- [ ] Code cleanup
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# ⚠️Help Wanted⚠️
How should I be thinking about testing this? I ran the existing apps and they seem ok. Since CRA helps with frontend in dev mode, deployment should not be a concern, right? Anything else to consider/test/validate?

### CODE REVIEW

File: waspc/src/Wasp/Generator/WebAppGenerator.hs [hunk 1]
```diff
@@ -105,10 +105,7 @@ npmDepsForWasp _spec =
             ("react-dom", "^17.0.2"),
             ("react-query", "^3.39.2"),
             ("react-router-dom", "^5.3.3"),
-            ("react-scripts", "4.0.3"),
-            -- NOTE: We need to specify this exact version of `react-error-overlay` for use with
-            -- `react-scripts` v4 due to this issue: https://github.com/facebook/create-react-app/issues/11773
-            ("react-error-overlay", "6.0.9")
+            ("react-scripts", "5.0.1")
```
Comment (shayneczyzewski) [inline waspc/src/Wasp/Generator/WebAppGenerator.hs:108]: Should we pin to exact version, or allow patch updates?

Comment (sodic) [reply waspc/src/Wasp/Generator/WebAppGenerator.hs:108]: Hm, we had it pinned before, but I wasn't sure why so I checked the history. It has been pinned down from [the very start](https://github.com/wasp-lang/wasp/commit/b43bfdfbfeba0e170061ed23bc8c2260fc56bb2e#diff-1303ebcff061f33c31aef11b7b7422d3826ffdabebbaad0a6be281b9051ee54c). Whenever we'd update a version, we'd leave it pinned. I wasn't sure why so I created an empty project with create-react-app and, sure enough, they pin it down too (I'm guessing Martin just copied what CRA generated at the time:
```json
  "dependencies": {
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.4.0",
    "@testing-library/user-event": "^13.5.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-scripts": "5.0.1",
    "web-vitals": "^2.1.4"
  },
```
Anyway, I think it's best we keep it pinned down.

**Review by sodic — APPROVED** [review]
LGTM!

**Comment by sodic** [comment]
> How should I be thinking about testing this? I ran the existing apps and they seem ok. Since CRA helps with frontend in dev mode, deployment should not be a concern, right? Anything else to consider/test/validate?

If the apps work (with both compile and start), I think we're good.

File: waspc/src/Wasp/Generator/WebAppGenerator.hs (hunk 1 above)
Comment (Martinsos) [reply waspc/src/Wasp/Generator/WebAppGenerator.hs:108]: Yup, I must have just kept it as it was.
Hm, I don't know! I did quick googling and it is unclear. We pin down, we avoid possible hiccups. We don't pin down, we allows for possible fixes. Let's leave it pinned down I guess, if that is what they recommended.

**Review by Martinsos — APPROVED** [review]

**Comment by Martinsos** [comment]
> > How should I be thinking about testing this? I ran the existing apps and they seem ok. Since CRA helps with frontend in dev mode, deployment should not be a concern, right? Anything else to consider/test/validate?
> 
> If the apps work (with both compile and start), I think we're good.

Ah sorry forgot to comment on this: yup I agree with @sodic , that should be enough of testing for now!

File: waspc/src/Wasp/Generator/WebAppGenerator.hs (hunk 1 above)
Comment (shayneczyzewski) [reply waspc/src/Wasp/Generator/WebAppGenerator.hs:108]: Sounds good, thanks for confirming gang. Will leave it as-is. 👍🏻

**Comment by shayneczyzewski** [comment]
Thanks for the review here @sodic and @Martinsos. I forgot to add the Changelog update for the possible breaking change (I will link to the CRA4 to 5 transition doc). After that, I will merge! 🚀

**Pull request merged**